src/providers/trusted_service/error.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use super::context::error::{Error, RpcCallerError, WrapperError};
+use parsec_interface::requests::ResponseStatus;
+
+impl From<RpcCallerError> for ResponseStatus {
+    fn from(error: RpcCallerError) -> Self {
+        match error {
+            RpcCallerError::EndpointDoesNotExist
+            | RpcCallerError::InvalidOpcode
+            | RpcCallerError::SerializationNotSupported
+            | RpcCallerError::ResourceFailure
+            | RpcCallerError::NotReady
+            | RpcCallerError::InvalidTransaction
+            | RpcCallerError::Internal => ResponseStatus::PsaErrorCommunicationFailure,
+            RpcCallerError::InvalidRequestBody
+            | RpcCallerError::InvalidResponseBody
+            | RpcCallerError::InvalidParameter => ResponseStatus::PsaErrorInvalidArgument,
```

Comment:
Couldn't these also be converted to `CommunicationFailure`? As in if they happen, isn't it a problem in Parsec rather than a wrong argument sent by the user?

Reply:
No, that was something that I was seeing in the test I was fixing - well, maybe I can move `InvalidResponseBody` and `InvalidParameter`, though I'm not sure what exactly the RPC client considers as "bad parameter". In that test, it was a "malformed" hash buffer that the RPC client didn't like, so it wasn't our fault.

Reply:
> a "malformed" hash buffer that the RPC client didn't like

malformed for PSA Crypto as well? As in if it is legal for the PSA Crypto interface but not for the RPC client maybe we should not return `InvalidArgument` and this is a bug in the client?

Reply:
Maybe, I'll look into it

Reply:
ah, I see what's happening - in that test we rely on the fact that a hash of 255 bytes won't be accepted as valid (which makes sense, a call to PSA crypto directly returns `InvalidArgument`), but the RPC client that talks to the TS needs to deserialize the call and has a limit of how large the hash can be. If the hash is longer than 64 bytes, it returns "invalid request body" so that one probably makes sense to convert to `PsaErrorInvalidArgument`. I'll move `RpcCallerError::ResponseBody` and `RpcCallerError::InvalidParameter` to `CommunicationError`.